File: fixtures/scale/typescript/small/align.ts
export class Align {
  static left(line: string, width: number): string {
    return line.padEnd(width);
  }

  static right(line: string, width: number): string {
    return line.padStart(width);
  }

  static center(line: string, width: number): string {
    const gap = Math.max(0, width - line.length);
    const left = Math.floor(gap / 2);
    return ' '.repeat(left) + line + ' '.repeat(gap - left);
  }
}

File: fixtures/scale/elixir/small/lib/style/align.ex
defmodule Style.Align do
  def align(:left, w, s), do: String.pad_trailing(s, w)

  def align(:right, w, s), do: String.pad_leading(s, w)

  def align(:center, w, s) do
    gap = max(w - String.length(s), 0)
    String.duplicate(" ", div(gap, 2)) <> s
  end
end
